Fix for kernel 6.8.0 changes

grillo-delmal opened the pull request:
I was testing Fedora 40 on my machine and I stumbled on a couple of compile problems with this module. This problems are the same that were reported by #7 , so this PR should fix them.

1) The linker can't find the `strlcpy` function anymore. Researching I found out that it seems that it was removed [0][1]. others sources I found about it[2] favor using `strscpy` instead, so I went for that to fix it.

2) The `vb2_queue` struct parameter `min_buffers_needed` [3] doesn't exist anymore, it sees that it's name was changed for `min_queued_buffers` [4]

Making these changes make it compile and work on my env.

[0] https://review.lttng.org/c/lttng-modules/+/11701
[1] https://www.debugpoint.com/linux-kernel-6-8/
[2] https://lwn.net/Articles/905777/
[3] https://elixir.bootlin.com/linux/v6.7.10/source/include/media/videobuf2-core.h#L549
[4] https://elixir.bootlin.com/linux/v6.8/source/include/media/videobuf2-core.h#L549

aquova commented:
Does this work on older kernel versions? Everything is working correctly now on 6.8, but I noticed now 6.6 LTS throws some errors when it attempts to build it.

grillo-delmal commented:
> Does this work on older kernel versions? Everything is working correctly now on 6.8, but I noticed now 6.6 LTS throws some errors when it attempts to build it.

Honestly, I didn't thought about it when I implemented the patch... now I added a conditional statement around the changed API code so that it uses the correct implementation depending on kernel version.

I doubt that the strscpy change would need to be encapsulated too, but feel free to test :)

GloriousEggroll commented:
thanks!
